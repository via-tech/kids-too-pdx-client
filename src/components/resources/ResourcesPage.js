import React from 'react';
import PropTypes from 'prop-types';
import styles from './Resources.css';

function ResourcesPage({ orgs }) {
  const orgsList = orgs.map((org, ind) =>
    <section key={ind}>
      <img alt={`${org.name} Logo`} src={org.logo} />
      <h2>{org.name}</h2>
      <p>{org.bio}</p>
    </section>
  );

  return (
    <section className={styles.Resources}>
      {orgsList}
    </section>
  );
}

ResourcesPage.propTypes = {
  orgs: PropTypes.array
};

export default ResourcesPage;
