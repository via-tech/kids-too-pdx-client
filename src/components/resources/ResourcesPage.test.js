import React from 'react';
import renderer from 'react-test-renderer';
import ResourcesPage from './ResourcesPage';

describe('Resources', () => {
  const props = {
    orgs: [
      {
        _id: '0',
        role: 'org',
        username: 'org1',
        password: 'org1pass',
        name: 'The Org1',
        email: 'org1@email.com',
        phone: '555-123-4567',
        address: {
          street: '123 Main St.',
          city: 'Portland',
          zipcode: '97203'
        },
        logo: 'org.com',
        bio: 'The bestest organization ever!',
        token: '0'
      },
      {
        _id: '1',
        role: 'org',
        username: 'org2',
        password: 'org2pass',
        name: 'The Org2',
        email: 'org2@email.com',
        phone: '555-123-4567',
        address: {
          street: '123 Main St.',
          city: 'Portland',
          zipcode: '97203'
        },
        logo: 'org.com',
        bio: 'The bestest organization ever!',
        token: '1'
      },
      {
        _id: '2',
        role: 'org',
        username: 'org3',
        password: 'org3pass',
        name: 'The Org3',
        email: 'org3@email.com',
        phone: '555-123-4567',
        address: {
          street: '123 Main St.',
          city: 'Portland',
          zipcode: '97203'
        },
        logo: 'org.com',
        bio: 'The bestest organization ever!',
        token: '2'
      }
    ]
  };

  it('snapshots Resources', () => {
    const tree = renderer.create(
      <ResourcesPage {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
