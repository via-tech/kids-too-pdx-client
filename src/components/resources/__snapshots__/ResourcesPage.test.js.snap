// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources snapshots Resources 1`] = `
<section
  className="Resources"
>
  <section>
    <img
      alt="The Org1 Logo"
      src="org.com"
    />
    <h2>
      The Org1
    </h2>
    <p>
      The bestest organization ever!
    </p>
  </section>
  <section>
    <img
      alt="The Org2 Logo"
      src="org.com"
    />
    <h2>
      The Org2
    </h2>
    <p>
      The bestest organization ever!
    </p>
  </section>
  <section>
    <img
      alt="The Org3 Logo"
      src="org.com"
    />
    <h2>
      The Org3
    </h2>
    <p>
      The bestest organization ever!
    </p>
  </section>
</section>
`;
